src/extension.ts

```diff
@@ -410,6 +411,24 @@ class DeviceTreeItem extends vscode.TreeItem {
         }
     }
 
+    public async printSysinfo() {
+        const messageBarItem = new StatusBarProgressionMessage("Grabbing ev3dev system info...");

```

Comment:
(You probably won't like this, but...) I think we can use use built-in `vscode.window` functions instead of `StatusBarProgressionMessage`. For the progress part, there is `vscode.window.withProgress()` and for the finish part, `vscode.window.setStatusBarMessage()` takes a timeout parameter.

The advantage of using `withProgress` is that it shows a spinner to indicate busy.

You can see an example of how I implemented this in the `download` function.

Here, you could do...

```javascript
        const sysinfo = await vscode.window.withProgress({
            location: vscode.progressLocation.Window,
            title: 'Grabbing system info...'
        }, progress => this.device.getSystemInfo());
        ...
        vscode.window.setStatusBarMessage('System info downloaded', 5000);
```

Reply:
I have no problem with using the progress item; I created that wrapper to remove the hard-coded time constants and such. If we want to use the progress version instead, I still want this to be wrapped in some sort of helper so we aren't embedding arbitrary times in our command implementations.

Reply:
What if we just define a global timeout constant that we use everywhere?

Reply:
```typescript
export function toastStatusBarMessage(message: string): void {
    vscode.window.setStatusBarMessage(message, 5000);
}
```

would be OK too.

I don't see anything that needs wrapped in `withProgress()`.